controller/aks-cluster-config-handler.go

```diff
@@ -665,6 +665,10 @@ func BuildUpstreamClusterState(ctx context.Context, secretsCache wranglerv1.Secr
 	// set addon monitoring profile
 	if addonProfile["omsAgent"] != nil {
 		upstreamSpec.Monitoring = addonProfile["omsAgent"].Enabled
+
+		if addonProfile["omsAgent"].Config == nil || len(addonProfile["omsAgent"].Config) == 0 {
+			return nil, fmt.Errorf("cannot set OMS Agent configuration retrieved from Azure")
```

Comment:
Can we have a more descriptive error here? Why are we getting a misconfigured response from Azure?

Reply:
Originally, I had 'cannot set invalid OMS agent configuration on cluster <cluster name>' but @dnoland1 requested the message above because we don't know if it's an invalid profile configuration. It's just a configuration that's nil or that we don't expect.

I'm not sure why the customer got a misconfigured OMS profile from Azure. We don't have enough info from them to know, we just know that whatever profile got returned didn't have .Config set.